### PR TITLE
Namespace support in prometheus rule

### PR DESCRIPTION
**Pull request by yjagdale**
Added support for namespace in the Prometheus rules. The original PR was #2694

### CODE REVIEW

**Comment by yjagdale** [comment]
Hey @Pothulapati, @romange  - Please have a look at this one. Sorry for the late response to PR. 

--Yash

**Comment by yjagdale** [comment]
Fixed linting issues for both files.

File: contrib/charts/dragonfly/ci/prometheusrules-values.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+serviceMonitor:
+  enabled: true
+prometheusRule:
+  enabled: true
```
Comment (Pothulapati) [inline contrib/charts/dragonfly/ci/prometheusrules-values.yaml:4]: Can you also add the namespace to verify the same? 🤔

Comment (yjagdale) [reply contrib/charts/dragonfly/ci/prometheusrules-values.yaml:4]: Done! Thanks! (It still doesn't update the golden file as the main constraint ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) doesn't return true on my local)

**Review by Pothulapati — APPROVED** [review]

**Comment by romange** [comment]
@Pothulapati  can I merge it?

**Comment by Pothulapati** [comment]
Okay! Not sure why this got closed, I just tried pushing a commit. Will raise a new PR

**Comment by yjagdale** [comment]
Can I raise new PR from yjagdale:main to main.

**Comment by Pothulapati** [comment]
@yjagdale Didn't want to waste more of your time. Raised it, and got it merged! sorry for the late review, and laso the PR getting closed! :/ Thanks for your effort.

**Pull request closed without merging**